Sources/PackageModel/ResolvedToolset.swift

```diff
@@ -0,0 +1,482 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Basics
+
+import class TSCBasic.Process
+
+#if os(Windows)
+private let hostExecutableSuffix = ".exe"
+#else
+private let hostExecutableSuffix = ""
+#endif
+
+public struct ResolvedToolset {
+    typealias SwiftCompilers = (compile: AbsolutePath, manifest: AbsolutePath)
+
+    public struct ToolProperties {
+        public let path: AbsolutePath
+        public let extraCLIOptions: [String]
+    }
+
+    public let effectiveTriple: Triple
```

Comment:
Is this a target triple or a host triple? I'd prefer the name to be explicit about that

Reply:
It is intentionally not explicit, because we need one resolved toolset per triple. I am envisioning that a higher level entity will keep track of per-triple resolved toolsets and it will know whether it is a host or target resolved toolset.

Ideally, I'd like resolved toolset to not even be aware of triples, but unfortunately that isn't possible since multiple of the tools are found using a triple.

Reply:
> It is intentionally not explicit, because we need one resolved toolset per triple.

But is that one resolved toolset per the host triple or the target triple?

> a higher level entity will keep track of per-triple resolved toolsets and it will know whether it is a host or target resolved toolset.

Sorry, I'm a bit confused, what does a "host or target resolved toolset" mean in this context? As I understand it, a toolset always needs to a pair of host and target triples to fully describe it.

Reply:
Oh I see, I guess I misunderstood your question. So yes, the triple would always be the target triple. My assumption was that the `toolset` that is being passed in is already known to be compatible with the host and all the computation inside the type is only looking for "installed" tools which are also assumed to always be compatible with the host.

Reply:
Ok, that makes sense. I'd still prefer this be intentionally explicit for readability. Even if it's obvious within `ResolvedToolset` that a triple is always the target triple, I think we had too many instances of SwiftPM code using `triple` name without being explicit if it's host or target and it was confusing to read that code after the fact.

Reply:
Yah, that makes sense, we can call this `targetTriple`.